modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/address/storage/AddressStorage.scala

```diff
@@ -1,67 +1,32 @@
 package org.tessellation.dag.l1.infrastructure.address.storage
 
-import cats.effect.MonadCancelThrow
+import cats.effect.Async
+import cats.effect.kernel.Ref
 import cats.syntax.functor._
-import cats.syntax.traverse._
 
 import org.tessellation.dag.l1.domain.address.storage.AddressStorage
-import org.tessellation.dag.l1.infrastructure.db.Database
 import org.tessellation.schema.address.Address
 import org.tessellation.schema.balance.Balance
 
-import doobie.implicits._
-import doobie.quill.DoobieContext
 import eu.timepit.refined.auto._
-import io.getquill.context.Context
-import io.getquill.{Literal, SqliteDialect}
 
 object AddressStorage {
-  case class StoredAddress(address: Address, balance: Balance)
+  def make[F[_]: Async]: F[AddressStorage[F]] =
+    Ref.of[F, Map[Address, Balance]](Map.empty).map(make(_))
 
-  type Ctx = DoobieContext.SQLite[Literal] with Context[SqliteDialect, Literal]
-
-  def make[F[_]: MonadCancelThrow: Database]: AddressStorage[F] =
-    make[F](new DoobieContext.SQLite[Literal](Literal))
-
-  def make[F[_]: MonadCancelThrow](ctx: Ctx)(implicit db: Database[F]): AddressStorage[F] =
+  def make[F[_]: Async](balances: Ref[F, Map[Address, Balance]]): AddressStorage[F] =
     new AddressStorage[F] {
-      val xa = db.xa
-
-      import ctx._
-
-      val getAddresses = quote {
-        querySchema[StoredAddress]("address")
-      }
-
-      val getAddressBalance = quote { (address: Address) =>
-        getAddresses.filter(_.address == address).map(_.balance).take(1)
-      }
-
-      val insertAddressBalance = quote { (address: Address, balance: Balance) =>
-        getAddresses
-          .insert(_.address -> address, _.balance -> balance)
-      }
-
-      val updateAddressBalance = quote { (address: Address, balance: Balance) =>
-        getAddresses
-          .filter(_.address == address)
-          .update(_.balance -> balance)
-      }
-
       def getBalance(address: Address): F[Balance] =
-        run(getAddressBalance(lift(address))).map(_.headOption.getOrElse(Balance.empty)).transact(xa)
+        balances.get.map(_.get(address).getOrElse(Balance.empty))
 
       def updateBalances(addressBalances: Map[Address, Balance]): F[Unit] =
-        addressBalances.toList.traverse {
-          case (address, balance) =>
-            run(getAddressBalance(lift(address)))
-              .map(_.headOption)
-              .flatMap {
-                case Some(_) => run(updateAddressBalance(lift(address), lift(balance)))
-                case None    => run(insertAddressBalance(lift(address), lift(balance)))
-              }
-        }.transact(xa).as(())
-
-      def clean: F[Unit] = run(getAddresses.delete).transact(xa).as(())
+        balances.update { m =>
+          addressBalances.foldLeft(m) {
```

Comment:
Isn't this the same as just `++` operator?

Reply:
tbh I don't know but currently it looks clear imo. `++` may be ambiguous whether we combine balances or update them

Reply:
@tbekas you're right. I'm changing it rn